sbt_common/internal_model/src/main/scala/uk/ac/wellcome/models/transformable/sierra/SierraRecordNumber.scala

```diff
@@ -0,0 +1,99 @@
+package uk.ac.wellcome.models.transformable.sierra
+
+import cats.syntax.either._
+import io.circe.{Decoder, Encoder}
+
+/** Represents the seven-digit IDs that are used to identify resources
+  * in Sierra.
+  *
+  * Implementation note: this is deliberately a class, not a case class,
+  * for two reasons:
+  *
+  *  1. It makes the [[s]] field private -- callers have to use the methods
+  *     [[withCheckDigit()]] and [[withoutCheckDigit()]].
+  *  2. It forces callers to consistently load the Circe decoder/encoder,
+  *     rather than relying on automatic case class derivations.  We _need_
+  *     to use the custom decoder when getting bib data from the Sierra API,
+  *     and then it would be good to be consistent internally.
+  *
+  */
+class SierraRecordNumber(s: String) {
```

Comment:
This could have a second parameter that is the record type: 
```
class SierraRecordNumber(s: String, recrodType: SierraRecordTypes.Value)
```
This way you would avoid passing the type to the `withCheckDigit` method, which feels a bit weird anyway, cause a bib number is always a bib number and same for an item number. You shouldn't be able to write a bib number prefixed with an i, which you can with this code

Reply:
Hmm. But when you're decoding a case class, how do you know if you're a bib record or an item record?

I feel a subclassing API coming on, I'll have a look…

Reply:
Ah I see, Yeah they become two different record numbers a that point

Reply:
Actually, I'm not sure that you would need two record number types. You could just a `recordType` field when encoding and then use that when decoding. You would need an encoder and decoder for `SierraRecordTypes`, probably similar to the one we have for `Licence`. Anyway, it should probably be in a separate PR

Reply:
It’s so disruptive, I’d rather get it right once than push a broken version that immediately needs patching.